torch/_dynamo/variables/builder.py

```diff
@@ -993,13 +993,21 @@ def wrap_literal(self, value):
                 guards=self.make_guards(GuardBuilder.CONSTANT_MATCH),
             )
 
+    def assert_not_wrapped_by_this_graph(self, value: torch.Tensor):
+        if self.tx.output.fake_mode.is_our_fake(value):
```

Comment:
Actually, this is a bit too strong. We actually require the recursive style of checking as per `is_fake`. So we want `is_fake and all fakes are created by the given fake mode`

Reply:
We will handle this as `is_fake` and `maybe_get_fake_mode(value) is self.tx.fake_mode`